core/CMakeLists.txt

```diff
@@ -0,0 +1,144 @@
+# Build libdronecore
+
+# This finds thread libs on Linux, Mac, and Windows.
+find_package(Threads REQUIRED)
+
+include_directories(
+    SYSTEM ${CMAKE_SOURCE_DIR}/libs/include
+)
+
+if (ANDROID)
+    include_directories(curl-android-ios/prebuilt-with-ssl/android/include)
+endif()
+
+if (IOS)
+    include_directories(curl-android-ios/prebuilt-with-ssl/ios/include)
+endif()
+
+if (IOS OR ANDROID OR MSVC OR APPLE)
+    set(library_type "STATIC")
+
+    # We need tinyxml2 for the camera definition parsing.
+    # We use the submodule when linking statically.
+    add_subdirectory(libs/tinyxml2 EXCLUDE_FROM_ALL)
```

Comment:
@julianoes: Why not having `libtinyxml2` statically linked for everybody, even Linux? That would simplify the build system, and I don't think it hurts (anyway it is statically linked for all the others).

Reply:
Right, why not.